File: leetcode/JavaScript实现/100-199题/0125.js
// 自己的代码
// 第一想法就是将除字母和数字以外的字符过滤掉 ，用正则非常好。
var isPalindrome = function(s) {
    if(s.length === 0)
        return true;
    let toUpper = s.toUpperCase().replace(/[^A-Za-z0-9]/g, "");
    let len = toUpper.length;
    for(let i = 0; i < len / 2; ++i) {
        if(toUpper[i] !== toUpper[len - i - 1])
            return false;
    }
    return true;
};

File: leetcode/C++实现/100-199题/0125.cpp
class Solution {
public:
    bool isPalindrome(string s) {
         if(s.length()==0)
            return true;
        string a;
        int s_size=s.size();
        for( int i=0;i<s_size;++i)
        {
            if( isdigit(s[i]) || isalpha(s[i]) )
            {
                a+=s[i];
            }
        }
        transform(a.begin(),a.end(),a.begin(),::toupper);
        int a_size=a.size();
        for( int i=0;i<=a_size/2;++i)
        {
            if( a[i]!=a[a_size-i-1] )
            {
                return false;
            }
        }
        return true;
    }
};